vcx/libvcx/src/disclosed_proof.rs

```diff
@@ -75,35 +74,191 @@ pub struct RequestedCreds {
+                                Some(cached_rev_state.timestamp)
+                            },
+                            _ => None
+                        };
+
+                        let (_, rev_reg_def_json) = get_rev_reg_def_json(&rev_reg_id)
+                            .map_err(|e| ProofError::CommonError(e))?;
+
+                        let (rev_reg_id, rev_reg_delta_json, timestamp) = get_rev_reg_delta_json(
+                            &rev_reg_id,
+                            from,
+                            to
+                        ).map_err(|e| ProofError::CommonError(e))?;
+
+                        let rev_state_json = anoncreds::libindy_prover_update_revocation_state(
+                            &rev_reg_def_json,
+                            &cached_rev_state.value,
+                            &rev_reg_delta_json,
+                            &cred_rev_id,
+                            &tails_file
+                        ).map_err(|e| ProofError::CommonError(e))?;
+
+                        if timestamp > cached_rev_state.timestamp {
+                            let new_cache = RevRegCache {
+                                rev_state: Some(RevState {
+                                    timestamp: timestamp,
+                                    value: rev_state_json.clone()
+                                })
+                            };
+                            set_rev_reg_cache(&rev_reg_id, &new_cache);
+                        }
+
+                        (rev_state_json, timestamp)
+                    }
+                } else {
+                    let (_, rev_reg_def_json) = get_rev_reg_def_json(&rev_reg_id)
+                        .map_err(|e| ProofError::CommonError(e))?;
+
+                    let (rev_reg_id, rev_reg_delta_json, timestamp) = get_rev_reg_delta_json(
+                        &rev_reg_id,
+                        None,
+                        to
+                    ).map_err(|e| ProofError::CommonError(e))?;
+
+                    let rev_state_json = anoncreds::libindy_prover_create_revocation_state(
+                        &rev_reg_def_json,
+                        &rev_reg_delta_json,
+                        &cred_rev_id,
+                        &tails_file
+                    ).map_err(|e| ProofError::CommonError(e))?;
+
+                    let new_cache = RevRegCache {
+                        rev_state: Some(RevState {
+                            timestamp: timestamp,
+                            value: rev_state_json.clone()
+                        })
+                    };
+                    set_rev_reg_cache(&rev_reg_id, &new_cache);
+
+                    (rev_state_json, timestamp)
+                };
+
+                let rev_state_json: Value = serde_json::from_str(&rev_state_json)
+                    .or(Err(ProofError::InvalidJson()))?;
+
+                rtn[rev_reg_id.to_string()] = json!({timestamp.to_string(): rev_state_json});
```

Comment:
In general it's possible situation that Proover should prepare multiply state of same revocation registry for different timestamp. But it's acceptable to merge first version without this functionality for MVP quality.

Reply:
@dkulic could you create a ticket for this technical debt and add TODO and comment into code?

Reply:
Sure.